source/VMelnalksnis.PaperlessDotNet/Documents/DocumentClient.cs

```diff
@@ -109,6 +110,23 @@ public async IAsyncEnumerable<Document<TFields>> GetAll<TFields>(int pageSize, [
 		return await GetCore<Document<TFields>>(id, cancellationToken).ConfigureAwait(false);
 	}
 
+
+	/// <inheritdoc />
+	public async Task<DocumentContent> Download(int id, bool original, CancellationToken cancellationToken = default)
+	{
+		var response = await _httpClient.GetAsync(original ? Routes.Documents.DownloadUri(id).ToString() :
+															 Routes.Documents.DownloadOriginalUri(id).ToString());
+
+		await response.EnsureSuccessStatusCodeAsync().ConfigureAwait(false);
+
+		var headers = response.Content.Headers;
+
+		return new DocumentContent(
+			content: await response.Content.ReadAsStreamAsync().ConfigureAwait(false),
+			fileName: headers.ContentDisposition?.FileName ?? string.Empty,
```

Comment:
Maybe we should just expose `ContentDisposition` and `ContentType` headers in `DocumentContent`, so that the user has access to all the properties?

Reply:
So to be clear: you mean that we remove the `FileName` and `ContentType` string-properties from `DocumentContent` and replace it with the properties `ContentDisposition` of type `ContentDispositionHeaderValue?` and `ContentType` of type `MediaTypeHeaderValue?`, right?

Reply:
Yes, the only thing I'm not sure on is nullability - the documentation says that they will be returned, so an option would be to make the properties on `DocumentContent` non-null, and either suppress the warnings or throw an exception. If all/most of the properties on `HeaderValue` types are nullable as well then it really does not matter, and you can leave the types as `ContentDispositionHeaderValue?` and `MediaTypeHeaderValue?`.

Reply:
I added the `MediaTypeHeaderValue` as not nullable because was always set when downloading the document, preview or thumbnail. The `ContentDispositionHeaderValue` was null when downloading the thumbnail, so i left that as nullable. When downloading the document this header was always set.

I am not sure if this is the best solution. Another option I thought of would be to create a separate result class for the download of the thumbnail which contains only the `Stream` and the `MediaTypeHeaderValue` so the user directly sees it he can expect to get the `ContentDispositionHeaderValue`.

What do you think of that?

Reply:
If the documentation says it will be set then should be ok.

A separate type sounds ok, but it will mean that you can't use the same core implementation for thumbnail then. I don't have a strong opinion either way on this, so go with whatever works best for you and your use case .

Reply:
Then I think this should be fine for now.